openfl/utilities/dataloading.py

```
@@ -41,47 +40,86 @@ def get_dataloader(
+
+    Returns:
+        DataLoader: A dataloader configured for the specified collaborator
+    """
     collaborator_names = list(plan.cols_data_paths)
-    collatorators_count = len(collaborator_names)
+    collaborator_count = len(collaborator_names)
 
-    if collaborator_index >= collatorators_count:
+    if collaborator_index >= collaborator_count:
         raise Exception(
-            f"Unable to construct full dataloader from collab_index={collaborator_index} "
-            f"when the plan has {collatorators_count} as total collaborator count. "
+            f"Unable to construct dataloader for index={collaborator_index} "
+            f"when the plan has {collaborator_count} total collaborators. "
             f"Please check plan/data.yaml file for current collaborator entries."
         )
 
     collaborator_name = collaborator_names[collaborator_index]
-    collaborator_data_path = plan.cols_data_paths[collaborator_name]
+    data_path = plan.cols_data_paths[collaborator_name]
 
-    # use seed_data provided by data_loader config if available
-    if "seed_data" in plan.config["data_loader"]["settings"] and not os.path.isdir(
-        collaborator_data_path
-    ):
-        os.makedirs(collaborator_data_path)
-        sample_data_zip_file = plan.config["data_loader"]["settings"]["seed_data"]
-        with zipfile.ZipFile(sample_data_zip_file, "r") as zip_ref:
-            zip_ref.extractall(collaborator_data_path)
+    # Handle seed data if provided in the plan
+    if "seed_data" in plan.config["data_loader"]["settings"]:
+        seed_data_zip = plan.config["data_loader"]["settings"]["seed_data"]
 
-    data_loader = plan.get_data_loader(collaborator_name)
+        if os.path.isfile(seed_data_zip):
+            os.makedirs(data_path, exist_ok=True)
+
+            # Always extract seed data when it's provided
+            # This ensures fresh data even if the directory already exists
+            with zipfile.ZipFile(seed_data_zip, "r") as zip_ref:
```

similar as above we should consider adding a check for valid zip file before attempting extraction

---

```suggestion
            # Check for corruption
            corrupt_file = zip_ref.testzip()
            if corrupt_file:
                logger.error(f"Corrupt file found in zip: {corrupt_file}")
                return False
           ... follow through the extraction flow
```

---

But with current way of implementation, it will throw the exception if file is corrupted and `fx plan initialize` command will fail. As this is first step of the process, we are failing fast. IMO, that should be ok, what do you think?